packages/aws-cdk/lib/util/asset-publishing.ts

```diff
@@ -50,7 +50,15 @@ class PublishingAws implements cdk_assets.IAws {
   }
 
   public async discoverCurrentAccount(): Promise<cdk_assets.Account> {
-    return (await this.sdk({})).currentAccount();
+    const account = await this.aws.defaultAccount();
+    return account ?? {
```

Comment:
Why do we return something here if we don't find a default account?

Reply:
:( because the contract of the function says it must return something. Not sure how much of an impact this is going to have... fairly sure this won't have any impact but I was trying to be safe and returning a dummy value and seeing what happens in real life.